Fix type errors due to typeless php docs

schlndh opened the pull request:
```
TypeError : SlevomatCodingStandard\Helpers\AnnotationTypeHelper::containsOneType(): Argument https://github.com/slevomat/coding-standard/pull/1 ($typeNode) must be of type PHPStan\PhpDocParser\Ast\Type\TypeNode, null given, called in SlevomatCodingStandard/Sniffs/TypeHints/ParameterTypeHintSniff.php on line 194

TypeError : SlevomatCodingStandard\Helpers\AnnotationTypeHelper::containsStaticOrThisType(): Argument https://github.com/slevomat/coding-standard/pull/1 ($typeNode) must be of type PHPStan\PhpDocParser\Ast\Type\TypeNode, null given, called in SlevomatCodingStandard/Helpers/AnnotationHelper.php on line 414
```

kukulich commented:
I'm sorry I don't understand what problem it solves.

schlndh commented:
@kukulich It fixes the type errors.

kukulich commented:
@schlndh Yes, but I need better description what code causes the `TypeError`.

This commit also does not look as bugfix: https://github.com/slevomat/coding-standard/pull/1389/commits/9c54c8f211b26fa7d1b75ed7d8a8134edb812fc2

schlndh commented:
```
$ git checkout master
$ git checkout fix-typeErrorsDueToTypelessPhpDocs -- tests/
$ bin/phing tests

...

There were 2 errors:

1) SlevomatCodingStandard\Sniffs\Commenting\UselessFunctionDocCommentSniffTest::testErrors
TypeError: SlevomatCodingStandard\Helpers\AnnotationTypeHelper::containsStaticOrThisType(): Argument #1 ($typeNode) must be of type PHPStan\PhpDocParser\Ast\Type\TypeNode, null given, called in SlevomatCodingStandard/Helpers/AnnotationHelper.php on line 410

SlevomatCodingStandard/Helpers/AnnotationTypeHelper.php:489
SlevomatCodingStandard/Helpers/AnnotationHelper.php:410
SlevomatCodingStandard/Sniffs/Commenting/UselessFunctionDocCommentSniff.php:87
vendor/squizlabs/php_codesniffer/src/Files/File.php:498
vendor/squizlabs/php_codesniffer/src/Files/LocalFile.php:92
SlevomatCodingStandard/Sniffs/TestCase.php:68
tests/Sniffs/Commenting/UselessFunctionDocCommentSniffTest.php:20

2) SlevomatCodingStandard\Sniffs\TypeHints\ParameterTypeHintSniffTest::testErrors
TypeError: SlevomatCodingStandard\Helpers\AnnotationTypeHelper::containsOneType(): Argument #1 ($typeNode) must be of type PHPStan\PhpDocParser\Ast\Type\TypeNode, null given, called in SlevomatCodingStandard/Sniffs/TypeHints/ParameterTypeHintSniff.php on line 194

SlevomatCodingStandard/Helpers/AnnotationTypeHelper.php:513
SlevomatCodingStandard/Sniffs/TypeHints/ParameterTypeHintSniff.php:194
SlevomatCodingStandard/Sniffs/TypeHints/ParameterTypeHintSniff.php:114
vendor/squizlabs/php_codesniffer/src/Files/File.php:498
vendor/squizlabs/php_codesniffer/src/Files/LocalFile.php:92
SlevomatCodingStandard/Sniffs/TestCase.php:68
tests/Sniffs/TypeHints/ParameterTypeHintSniffTest.php:24

ERRORS!
Tests: 1143, Assertions: 7113, Errors: 2.
```

The package versions I tested with are as follows (i.e. fresh as of now). The relevant package is `phpdoc-parser` which is version 1.6.3:

```
dealerdirect/phpcodesniffer-composer-installer v0.7.2  PHP_CodeSniffer Standards Compose...
doctrine/instantiator                          1.4.1   A small, lightweight utility to i...
myclabs/deep-copy                              1.11.0  Create deep copies (clones) of yo...
nikic/php-parser                               v4.14.0 A PHP parser written in PHP
phar-io/manifest                               2.0.3   Component for reading phar.io man...
phar-io/version                                3.2.1   Library for handling version info...
phing/phing                                    2.17.3  PHing Is Not GNU make; it's a PHP...
php-parallel-lint/php-parallel-lint            v1.3.2  This tool check syntax of PHP fil...
phpdocumentor/reflection-common                2.2.0   Common reflection classes used by...
phpdocumentor/reflection-docblock              5.3.0   With this component, a library ca...
phpdocumentor/type-resolver                    1.6.1   A PSR-5 based resolver of Class n...
phpspec/prophecy                               v1.15.0 Highly opinionated mocking framew...
phpstan/phpdoc-parser                          1.6.3   PHPDoc parser with support for nu...
phpstan/phpstan                                1.7.14  PHPStan - PHP Static Analysis Tool
phpstan/phpstan-deprecation-rules              1.0.0   PHPStan rules for detecting usage...
phpstan/phpstan-phpunit                        1.1.1   PHPUnit extensions and rules for ...
phpstan/phpstan-strict-rules                   1.2.3   Extra strict and opinionated rule...
phpunit/php-code-coverage                      9.2.15  Library that provides collection,...
phpunit/php-file-iterator                      3.0.6   FilterIterator implementation tha...
phpunit/php-invoker                            3.1.1   Invoke callables with a timeout
phpunit/php-text-template                      2.0.4   Simple template engine.
phpunit/php-timer                              5.0.3   Utility class for timing
phpunit/phpunit                                9.5.20  The PHP Unit Testing framework.
sebastian/cli-parser                           1.0.1   Library for parsing CLI options
sebastian/code-unit                            1.0.8   Collection of value objects that ...
sebastian/code-unit-reverse-lookup             2.0.3   Looks up which function or method...
sebastian/comparator                           4.0.6   Provides the functionality to com...
sebastian/complexity                           2.0.2   Library for calculating the compl...
sebastian/diff                                 4.0.4   Diff implementation
sebastian/environment                          5.1.4   Provides functionality to handle ...
sebastian/exporter                             4.0.4   Provides the functionality to exp...
sebastian/global-state                         5.0.5   Snapshotting of global state
sebastian/lines-of-code                        1.0.3   Library for counting the lines of...
sebastian/object-enumerator                    4.0.4   Traverses array structures and ob...
sebastian/object-reflector                     2.0.4   Allows reflection of object attri...
sebastian/recursion-context                    4.0.4   Provides functionality to recursi...
sebastian/resource-operations                  3.0.3   Provides a list of PHP built-in f...
sebastian/type                                 3.0.0   Collection of value objects that ...
sebastian/version                              3.0.2   Library that helps with managing ...
squizlabs/php_codesniffer                      3.7.1   PHP_CodeSniffer tokenizes PHP, Ja...
theseer/tokenizer                              1.2.1   A small library for converting to...
webmozart/assert                               1.11.0  Assertions to validate method inp...
```
-----------------
As for the last commit: I just felt that a type-less `@param` with no description etc. for an untyped parameter should also be considered useless. But that's just for consistency (now that I think of it, I should have probably also added a test case for `UselessFunctionDocCommentSniff` for this scenario - the two cases I added there have typed parameters). If you don't like it, I can remove it.

kukulich commented:
@schlndh I understand that there's a `TypeError`. I want to see the PHP code that causes the `TypeError` in Slevomat CS. It's not clear for me from the commit messages.

> I just felt that a type-less @param with no description etc. for an untyped parameter should also be considered useless.

Yes, you're right but it's BC break (can break builds) and should be in separate PR. It has to go to next major release.

schlndh commented:
@kukulich The type errors are caused by code like this:

```php

	/** @param $a */
	private function noTypeHintNoAnnotationWithPhpdoc($a)
	{
	}
	
	/**
	 * @param $a
	 */
	public function onlyParameterWithoutType(int $a): void
	{
	}
```

> Yes, you're right but it's BC break (can break builds) and should be in separate PR. It has to go to next major release.

Ok, I'll remove the commit when I get the chance.

kukulich commented:
@schlndh There is already this test and there's no `TypeError`

```php
	/**
	 * @param $a
	 */
	public function noType(int $a)
	{

	}
```

So is your second example valid?

schlndh commented:
@kukulich
>So is your second example valid?

Yes, but not for `ParameterTypeHintSniff` (the example you're referring to is from its test), because there it only happens for parameters without native typehint. Its TypeError happens [here](https://github.com/slevomat/coding-standard/blob/05a9c980646b0dc84777d29cdd2922ff25f4d3db/SlevomatCodingStandard/Sniffs/TypeHints/ParameterTypeHintSniff.php#L194), which is inside `foreach ($parametersWithoutTypeHint as $parameterName)`.

However, the TypeError for this case happens in `UselessFunctionDocCommentSniff`.

schlndh commented:
@kukulich Ok. Sorry about the confusion. I felt like the new test were enough of an explanation, but I guess that's just because I already knew what it was about. :see_no_evil: I removed the BC-breaking commit.
